### ISSUE TITLE
How to use S3Client AWS SDK v3 to upload thumbnails to S3

### ISSUE THREAD

**Issue by Aerendir** [liip/LiipImagineBundle#626]
I'm trying to upload the generated thumbnails to S3 using the `AwsS3Resolver`but I cannot configure it.

If I define a service for the Àws\Credentials` class, the string `service.awscredentials`is passed instead of the class.
If I pass directly an array, LiipImagine throws an error as the parameter type is incorrect. So, in the end, I cannot configure the resolver using the version 3 of the AWS SDK.

[I've wrote on StackOverflow a detailed exaplanation of what's happening](http://stackoverflow.com/questions/31789744/how-to-use-s3-aws-sdk-v3-to-cache-images-generated-by-liipimaginebundle).

Now, I'd like to understand if this is a fault of mine in configuring the bundle or it is a problem of LiipImagine that doesn't support the version 3 of the AWS SDK?

**Comment by makasim** [comment]
Have you checked this doc: http://symfony.com/doc/master/bundles/LiipImagineBundle/cache-resolver/aws_s3.html ?

**Comment by morgenstern** [comment]
I have the same problem .. @Aerendir did you find a solution? 

I tried the following settings:

```
# app/config/config.yml

liip_imagine:
    resolvers:
        default:
            web_path: ~
        s3resolver:
            aws_s3:
                cache: true
                proxies: [%cdn_base_url%]
# ------- talking about this section ------- 
                client_config:
                    credentials:
                        class: Aws\Credentials\Credentials
                        arguments: ["%aws_key%", "%aws_secret_key%"]
                    version:    %aws_sdk_version%
                    region:     %aws_region%
# -------------- end -------------- 
                bucket: %aws_bucket%
                get_options:
                    Scheme: 'https'
                put_options:
                    CacheControl: 'max-age=86400'
    filter_sets:
        cache: ~
        shop_logo_thumb:
            cache: s3resolver
            quality: 85
            filters:
                thumbnail: { size: [180, 90], mode: inset }
```

gives me this **error message:** 
> _InvalidTypeException in ScalarNode.php line 36:
> Invalid type for path "liip_imagine.resolvers.s3resolver.aws_s3.client_config.credentials". Expected scalar, but got array._

I also tried this:

```
                client_config:
                    credentials:
                        key: %aws_key%
                        secret: %aws_secret_key%
                    version:    %aws_sdk_version%
                    region:     %aws_region%
```
**with this result:**
> _InvalidTypeException in ScalarNode.php line 36:
> Invalid type for path "liip_imagine.resolvers.s3resolver.aws_s3.client_config.<CONTENTOFMY%aws_key%PARAMETER>". Expected scalar, but got array._

Any idea on how to fix this?

**Comment by Aerendir** [comment]
No, unfortunately I din't find a solution. But in the meantime I simply didn't use the feature. It is in the todo list...

**Comment by makasim** [comment]
fixed in 1.4.1
